v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/utils/convert-to-edr.ts

```diff
@@ -274,28 +281,32 @@ export async function normalizeEdrNetworkAccountsConfig(
 
 export function hardhatChainsToEdrChains(
   chains: EdrNetworkChainsConfig,
+  chainType: ChainType,
 ): ChainConfig[] {
-  const edrChains: ChainConfig[] = [];
-
-  for (const [chainId, hardforkConfig] of chains) {
-    const hardforks = [];
-
-    for (const [hardfork, blockNumber] of hardforkConfig.hardforkHistory) {
-      const specId = hardhatHardforkToEdrSpecId(getHardforkName(hardfork));
-
-      hardforks.push({
-        blockNumber: BigInt(blockNumber),
-        specId,
-      });
-    }
-
-    edrChains.push({
-      chainId: BigInt(chainId),
-      hardforks,
-    });
-  }
-
-  return edrChains;
+  return (
+    Array.from(chains)
+      // Skip chains that don't match the expected chain type
+      .filter(([_, config]) => {
+        if (chainType === GENERIC_CHAIN_TYPE) {
+          // When "generic" is requested, include both "generic" and "l1" chains
```

Comment:
Can you confirm this? If the passed `chainType` (i.e., the "resolved" chain type) is `"generic"`, we want to return both `"generic"` and `"l1"` hardforks from the `chains` map.

Reply:
Generic is a more permissive version of L1. Thus, all L1 one chains can be treated as generic chains.

Generic chains cannot be treated as L1 chains.

Reply:
So it should be the other way around? If "l1" is passed, we include both "generic" and "l1"? That would mean "generic" chains won’t pass any chains to EDR,  is that correct?

Reply:
No, I think it's correct as is.

E.g. A user can define a generic chain type for Arbitrum. In that case it should not be sent to an L1 chain.